services/wfm/src/main/java/org/openkilda/wfm/topology/reroute/bolts/RerouteBolt.java

```diff
@@ -0,0 +1,130 @@
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.reroute.bolts;
+
+import static java.lang.String.format;
+import static org.openkilda.messaging.Utils.MAPPER;
+
+import org.openkilda.messaging.command.CommandData;
+import org.openkilda.messaging.command.CommandMessage;
+import org.openkilda.messaging.command.reroute.RerouteAffectedFlows;
+import org.openkilda.messaging.command.reroute.RerouteFlows;
+import org.openkilda.messaging.command.reroute.RerouteInactiveFlows;
+import org.openkilda.messaging.info.event.PathNode;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.wfm.AbstractBolt;
+import org.openkilda.wfm.topology.reroute.StreamType;
+import org.openkilda.wfm.topology.reroute.service.RerouteService;
+
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+
+public class RerouteBolt extends AbstractBolt {
+
+    private static final Logger logger = LoggerFactory.getLogger(RerouteBolt.class);
+
+    public static final String FLOW_ID_FIELD = "flow-id";
+    public static final String CORRELATION_ID_FIELD = "correlation-id";
+
+    private PersistenceManager persistenceManager;
+    private transient RerouteService rerouteService;
+
+    public RerouteBolt(PersistenceManager persistenceManager) {
+        this.persistenceManager = persistenceManager;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        this.rerouteService = new RerouteService(repositoryFactory);
+        super.prepare(stormConf, context, collector);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void handleInput(Tuple tuple) {
+        String request = tuple.getString(0);
+
+        CommandMessage message;
+        try {
+            message = MAPPER.readValue(request, CommandMessage.class);
+        } catch (IOException e) {
+            logger.error("Error during parsing request for Reroute topology", e);
+            return;
+        }
+
+        CommandData commandData = message.getData();
+        if (commandData instanceof RerouteFlows) {
+
+            if (commandData instanceof RerouteAffectedFlows) {
```

Comment:
You do not use second level of "if"... so you can remove wrapping "if".

(minor)